src/shogun/preprocessor/RandomFourierGaussPreproc.cpp

```diff
@@ -8,393 +8,108 @@
-	    2 * pi, m_prng);
-	
-	UniformRealDistribution<float64_t> uniform_real_dist(-1.0, 1.0);
-	for (int32_t  i = 0; i < cur_dim_feature_space; ++i) {
-		for (int32_t k = 0; k < cur_dim_input_space; ++k) {
-			float64_t x1,x2;
-			float64_t s = 2;
-			while ((s >= 1) ) {
-				// Marsaglia polar for gaussian
-				x1 = uniform_real_dist(m_prng);
-				x2 = uniform_real_dist(m_prng);
-				s=x1*x1+x2*x2;
-			}
-
-			// =  x1/std::sqrt(val)* std::sqrt(-2*std::log(val));
-			randomcoeff_multiplicative[i * cur_dim_input_space + k] =
-			    x1 * std::sqrt(-2 * std::log(s) / s) / kernelwidth;
-		}
-	}
-
-	io::info("finished: initializing randomcoefficients ");
-
-	return true;
-}
-
-void RandomFourierGaussPreproc::get_randomcoefficients(
-		float64_t ** randomcoeff_additive2,
-		float64_t ** randomcoeff_multiplicative2, int32_t *dim_feature_space2,
-		int32_t *dim_input_space2, float64_t* kernelwidth2) const {
-
-	ASSERT(randomcoeff_additive2)
-	ASSERT(randomcoeff_multiplicative2)
-
-	if (!test_rfinited()) {
-		*dim_feature_space2 = 0;
-		*dim_input_space2 = 0;
-		*kernelwidth2=1;
-		*randomcoeff_additive2 = NULL;
-		*randomcoeff_multiplicative2 = NULL;
-		return;
-	}
-
-	*dim_feature_space2 = cur_dim_feature_space;
-	*dim_input_space2 = cur_dim_input_space;
-	*kernelwidth2=cur_kernelwidth;
-
-	*randomcoeff_additive2 = SG_MALLOC(float64_t, cur_dim_feature_space);
-	*randomcoeff_multiplicative2 = SG_MALLOC(float64_t, cur_dim_feature_space*cur_dim_input_space);
-
-	std::copy(randomcoeff_additive, randomcoeff_additive+cur_dim_feature_space,
-			*randomcoeff_additive2);
-	std::copy(randomcoeff_multiplicative, randomcoeff_multiplicative+cur_dim_feature_space*cur_dim_input_space,
-			*randomcoeff_multiplicative2);
-
-
-}
-
-void RandomFourierGaussPreproc::set_randomcoefficients(
-		float64_t *randomcoeff_additive2,
-		float64_t * randomcoeff_multiplicative2,
-		const int32_t dim_feature_space2, const int32_t dim_input_space2, const float64_t kernelwidth2) {
-	dim_feature_space = dim_feature_space2;
-	dim_input_space = dim_input_space2;
-	kernelwidth=kernelwidth2;
-
-	SG_FREE(randomcoeff_multiplicative);
-	randomcoeff_multiplicative=NULL;
-	SG_FREE(randomcoeff_additive);
-	randomcoeff_additive=NULL;
-
-	cur_dim_feature_space=dim_feature_space;
-	cur_dim_input_space = dim_input_space;
-	cur_kernelwidth=kernelwidth;
-
-	if( (dim_feature_space>0) && (dim_input_space>0) )
-	{
-	randomcoeff_additive=SG_MALLOC(float64_t, cur_dim_feature_space);
-	randomcoeff_multiplicative=SG_MALLOC(float64_t, cur_dim_feature_space*cur_dim_input_space);
-
-	std::copy(randomcoeff_additive2, randomcoeff_additive2
-			+ dim_feature_space, randomcoeff_additive);
-	std::copy(randomcoeff_multiplicative2, randomcoeff_multiplicative2
-			+ cur_dim_feature_space*cur_dim_input_space, randomcoeff_multiplicative);
-	}
+void RandomFourierGaussPreproc::init_basis(int32_t dim_input_space)
+{
+	io::info("Generating {} random coefficients for input space dimension {}", m_dim_output, dim_input_space);
```

Comment:
minor: think this line might be too long

Reply:
```io::info("Creating Fourier Basis Matrix {}x{}", m_dim_output, dim_input_space); ```